svf-llvm/lib/SVFIRBuilder.cpp

```diff
@@ -1398,7 +1398,37 @@ const Value* SVFIRBuilder::getBaseValueForExtArg(const Value* V)
         }
         if(totalidx == 0 && !SVFUtil::isa<StructType>(value->getType()))
             value = gep->getPointerOperand();
+    } else if (const LoadInst* load = SVFUtil::dyn_cast<LoadInst>(value)) {
+        // https://github.com/SVF-tools/SVF/issues/1650
+        /*!
+            @i1 = dso_local global %struct.interesting { i32 0, ptr @f1, ptr @f2 }
+            @n1 = dso_local global %struct.nested_ptr { i32 0, ptr @i1 }
+
+            %0 = load ptr, ptr getelementptr inbounds (%struct.nested_ptr, ptr @n1, i32 0, i32 1)
+            call void @llvm.memcpy.p0.p0.i64(ptr align 8 %interesting_stub, ptr align 8 %0, i64 24, i1 false)
+            the base value for %0 is @i1. load -> gep -> global
+        */
+        const Value* loadP = load->getPointerOperand();
+        if (const GetElementPtrInst* gep = SVFUtil::dyn_cast<GetElementPtrInst>(loadP)) {
+            APOffset totalidx = 0;
+            for (bridge_gep_iterator gi = bridge_gep_begin(gep), ge = bridge_gep_end(gep); gi != ge; ++gi)
+            {
+                if(const ConstantInt* op = SVFUtil::dyn_cast<ConstantInt>(gi.getOperand()))
+                    totalidx += LLVMUtil::getIntegerValue(op).first;
+            }
+            const Value * pointer_operand = gep->getPointerOperand();
+            if (auto *glob = SVFUtil::dyn_cast<GlobalVariable>(pointer_operand)) {
+                if (auto *initializer = llvm::dyn_cast<
+                    ConstantStruct>(glob->getInitializer())) {
```

Comment:
better to use the existing method to get the object (that SVF has collected) rather than identifying the global object and its initializer.